### PR TITLE
Outbox archiver

### PR DESCRIPTION
**Pull request by preardon**
The idea behind this is to have a service that can move old messages from the outbox to another provider

### CODE REVIEW

File: src/Paramore.Brighter.Archive.Azure/AzureBlobArchiveProvider.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+using Azure.Storage.Blobs;
+
+namespace Paramore.Brighter.Storage.Azure;
+
+public class AzureBlobArchiveProvider : IAmAnArchiveProvider
```
Comment (iancooper) [inline src/Paramore.Brighter.Archive.Azure/AzureBlobArchiveProvider.cs:5]: I think the main difference between this and the luggage store is that with the luggage store we only store the payload, not the whole message, hence our use of a stream

File: src/Paramore.Brighter.Extensions.Hosting/HostedServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -23,5 +23,19 @@ public static class HostedServiceCollectionExtensions
             brighterBuilder.Services.AddHostedService<TimedOutboxSweeper>();
             return brighterBuilder;
         }
+
+        public static IBrighterBuilder UseOutboxArchiver(this IBrighterBuilder brighterBuilder,
```
Comment (iancooper) [inline src/Paramore.Brighter.Extensions.Hosting/HostedServiceCollectionExtensions.cs:27]: I assume we think most folks will run this stand alone, perhaps with their sweeper?

File: src/Paramore.Brighter/NullOutboxArchiveProvider.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Paramore.Brighter.Logging;
+
+namespace Paramore.Brighter
+{
+    /// <summary>
+    /// Use this archiver will result in messages just being deleted from the outbox and not stored
+    /// </summary>
+    public class NullOutboxArchiveProvider : IAmAnArchiveProvider
```
Comment (iancooper) [inline src/Paramore.Brighter/NullOutboxArchiveProvider.cs:11]: LOL, send to null

**Review by iancooper — APPROVED** [review]

File: src/Paramore.Brighter.Archive.Azure/AzureBlobArchiveProvider.cs (hunk 1 above)
Comment (preardon) [reply src/Paramore.Brighter.Archive.Azure/AzureBlobArchiveProvider.cs:5]: Completely agree, it was an optional extra here but I thought it was a Nice Touch however it looks like azure doesn't want to give me access to do this 🤣

File: src/Paramore.Brighter.Extensions.Hosting/HostedServiceCollectionExtensions.cs (hunk 2 above)
Comment (preardon) [reply src/Paramore.Brighter.Extensions.Hosting/HostedServiceCollectionExtensions.cs:27]: That's the plan, I'm looking at putting this in the same process as outbox sweeper so I've got an "Outbox management" service

File: src/Paramore.Brighter/NullOutboxArchiveProvider.cs (hunk 3 above)
Comment (preardon) [reply src/Paramore.Brighter/NullOutboxArchiveProvider.cs:11]: Well I had to ways to go, either write code in to give you an "Action" or default register you a don't write store. On the fence weather I like it, just know I don't currently hate it, thoughts @iancooper

File: src/Paramore.Brighter.Extensions.Hosting/TimedOutboxArchiverOptions.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+namespace Paramore.Brighter.Extensions.Hosting
+{
+    public class TimedOutboxArchiverOptions
+    {
+        /// <summary>
+        /// The timer interval in hours.
```
Comment (preardon) [inline src/Paramore.Brighter.Extensions.Hosting/TimedOutboxArchiverOptions.cs:6]: Must confirm or change this to second

**Pull request merged**